R/USA.R

```diff
@@ -0,0 +1,131 @@
+    supported_region_codes = list("1" = "iso_3166_2",
+                                  "2" = "fips"),
+    #' @field data_url List of named links to raw data. Since there are no common
+    #' data sources for level 1 and level 2 for the USA, this list is empty.
+    data_url = list(),
+    #' @field data_url_level_1 List of named links to raw data for level 1;
+    #' not used or accessed when generating level 2 results.
+    data_url_level_1 = list(
+      "main" = "https://raw.githubusercontent.com/nytimes/covid-19-data/master/us-states.csv" # nolint
+    ),
+    #' @field data_url_level_2 link to raw data for level 2
+    data_url_level_2 = list(
+      "main" = "https://raw.githubusercontent.com/nytimes/covid-19-data/master/us-counties.csv" # nolint
+    ),
+    #' @field source_data_cols existing columns within the raw data
+    source_data_cols = c("cases_total", "deaths_total"),
+
+    #' @description Set up a table of region codes for clean data
+    #' @importFrom tibble tibble tribble
+    set_region_codes = function() {
+      self$codes_lookup$`1` <- usa_codes <- tibble::tibble(
+        level_1_region_code = c(
+          "US-AL", "US-AK", "US-AZ", "US-AR", "US-CA", "US-CO", "US-CT",
+          "US-DE", "US-FL", "US-GA", "US-HI", "US-ID", "US-IL", "US-IN",
+          "US-IA", "US-KS", "US-KY", "US-LA", "US-ME", "US-MD", "US-MA",
+          "US-MI", "US-MN", "US-MS", "US-MO", "US-MT", "US-NE", "US-NV",
+          "US-NH", "US-NJ", "US-NM", "US-NY", "US-NC", "US-ND", "US-OH",
+          "US-OK", "US-OR", "US-PA", "US-RI", "US-SC", "US-SD", "US-TN",
+          "US-TX", "US-UT", "US-VE", "US-VA", "US-WA", "US-WV", "US-WI",
+          "US-WY", "US-DC", "US-AS", "US-GU", "US-MP", "US-PR", "US-UM",
+          "US-VI"
+        ),
+        region = c(
+          "Alabama", "Alaska", "Arizona", "Arkansas", "California", "Colorado",
+          "Connecticut", "Delaware", "Florida", "Georgia", "Hawaii", "Idaho",
+          "Illinois", "Indiana", "Iowa", "Kansas", "Kentucky", "Louisiana",
+          "Maine", "Maryland", "Massachusetts", "Michigan", "Minnesota",
+          "Mississippi", "Missouri", "Montana", "Nebraska", "Nevada",
+          "New Hampshire", "New Jersey", "New Mexico", "New York",
+          "North Carolina", "North Dakota", "Ohio", "Oklahoma", "Oregon",
+          "Pennsylvania",  "Rhode Island", "South Carolina", "South Dakota",
+          "Tennessee", "Texas", "Utah", "Vermont", "Virginia", "Washington",
+          "West Virginia", "Wisconsin", "Wyoming", "District of Columbia",
+          "American Samoa", "Guam", "Northern Mariana Islands", "Puerto Rico",
+          "Minor Outlying Islands", "Virgin Islands"
+        )
+      )
+      self$codes_lookup$`2` <- self$codes_lookup$`1`
+    },
+
+    #' @description USA specific state level data cleaning
+    #'
+    clean = function() {
+      message_verbose(self$verbose, "Cleaning data")
+
+      if (self$level == "1") {
```

Comment:
We should really make this the default to reduce copy paste. Ideally doing it using the self level value and building the function names programmatically would be neat as we could add new levels without issue.